runtime/lua/vim/lsp/util.lua

```diff
@@ -1497,18 +1502,30 @@ do --[[ References ]]
   ---@param bufnr buffer id
   ---@param references List of `DocumentHighlight` objects to highlight
   ---@see https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#documentHighlight
-  function M.buf_highlight_references(bufnr, references)
+  function M.buf_highlight_references(bufnr, references, client_id)
     validate { bufnr = {bufnr, 'n', true} }
+    local client = vim.lsp.get_client_by_id(client_id)
+    if not client then
+      return
+    end
```

Comment:
Changing the function signature in a way that breaks functionality for every single existing caller (since client_id will be nil for all existing callers as it's a new parameter) is a pretty unnecessary landmine to hide.

This caused https://github.com/RRethy/vim-illuminate/issues/85.

A default value for the `offset_encoding` when `client_id` isn't present would make more sense IMO.